tests/bud.bats

```diff
@@ -117,6 +117,48 @@ symlink(subdir)"
   expect_output --substring $(realpath "${TESTSDIR}/bud/dockerignore3/.dockerignore")
 }
 
+# Following test must fail since we are trying to run linux/arm64 on linux/amd64
+# Issue: https://github.com/containers/buildah/issues/3712
+@test "build-with-inline-platform" {
+  # Host arch
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
+  for try in amd64 arm64 ppc64le s390x; do
+    if [[ "$otherarch" != "$myarch" ]]; then
+      otherarch="$myarch"
+      break
+    fi
+  done
+  # ...create a Containerfile with --platform=linux/$otherarch
+  cat > ${TESTSDIR}/bud/Dockerfile << _EOF
+FROM --platform=linux/$otherarch alpine
+RUN echo "hello"
+_EOF
+
+  run_buildah '?' build --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
+  if [[ $status -eq 0 ]]; then
+    # Build succeeded: we must have qemu-user-static installed. Confirm that
+    # the 'uname' emits what we expect
+    expect_output --from="${lines[-1]}" "$otherarch"
+    run_buildah inspect --format '{{ .OCIv1.Architecture }}' test
+    expect_output --substring "$otherarch"
+  else
+    # Build failed: we DO NOT have qemu-user-static installed.
+    expect_output --substring "format error"
+  fi
+}
+
+# Following test must pass since we want to tag image as arm64
+# Test for use-case described here: https://github.com/containers/buildah/issues/3261
+@test "build-with-inline-platform-amd-but-tag-as-arm" {
+  # Tries building image where baseImage has --platform=linux/arm64
+  run_buildah build --platform linux/arm64 --signature-policy ${TESTSDIR}/policy.json -t test -f ${TESTSDIR}/bud/platform/Dockerfileamd
```

Comment:
Wouldn't that break as well without qemu-user-static?  I don't see the "trying" as suggested by the comment.

Reply:
Only need user-static if you are doing a RUN command. (Having not checked if this does a RUN.)

Reply:
```
FROM --platform=linux/amd64 busybox
RUN uname -m
```

Reply:
ackd let me add same test here as well.

Reply:
I am removing `RUN` call so this does not fail.